crates/hyfetch/src/bin/hyfetch.rs

```diff
@@ -57,7 +57,19 @@ fn main() -> Result<()> {
     // Use a custom distro
     let distro = options.distro.as_ref();
 
-    let backend = options.backend.unwrap_or(Backend::Neofetch);
+    let backend = options.backend.unwrap_or({
+        if cfg!(all(unix, not(target_os = "android"))) {
+            Backend::Neofetch
+        } else {
+            // fastfetch is not a good fallback on *nix; it only supports FreeBSD and not
+            // other *BSD for example
```

Comment:
I see the concern of *BSD. However linux and darwin belong to *nix too. [Even for FreeBSD, fastfetch has much better support than neofetch.](https://github.com/fastfetch-cli/fastfetch/wiki#freebsd)

I don't get the comment here.

Reply:
Why `not(target_os = "android")`?

Reply:
> Why `not(target_os = "android")`?

To use fastfetch on Android ^^

Reply:
> However linux

Sorry, I'm confused. I chose to stick to neofetch for Linux because of your comment. Has the situation changed?

> Obtaining the linux distro name is somewhat complex. Obtaining the OS name (Linux, Windows, macOS, etc) should be easy. If a non-linux OS is detected, hyfetch should be able not to ask neofetch.

https://github.com/hykilpikonna/hyfetch/issues/300#issuecomment-2138888037

Reply:
It's about detecting Linux distro name. I can't recommend fastfetch in this case because the logos supported by hyfetch are all generated from neofetch source code. Since fastfetch is not neofetch, fastfetch may report different results for some distros which may confuse hyfetch. However it's a temp solution. In order not to slow down hyfetch, you may want to implement the distro detection logic in hyfetch.

For backend itself, I definitely recommand fastfetch since I'm the fastfetch author.

Reply:
@CarterLi Indeed, the current distro matching logic in hyfetch is not ideal. It'd be nice if there's a library for that lol

I think we should have a mapping of distro to named ascii art, and still support overriding with a specific named ascii art. They should be separate.

Reply:
Another option is that let backend generate ascii logo for hyfetch.

Reply:
@CarterLi too slow for neofetch... which I presume is why hyfetch is using generated code in the first place 🙈

Reply:
Anyway, I think the extra complexity / maintenance burden is not worth it. Users can easily select fastfetch themselves. And if something doesn't work right, perhaps we just handle those on a case-by-case basis.

But I think we can default to fastfetch on FreeBSD.

Reply:
UPDATE: Changed to just default to fastfetch if it's installed.

I think that's a reasonable thing to do? If the user has installed fastfetch, they most likely want to use it. And if they run into a distro / ascii name mismatch, they can report so that we can add a workaround or something...

Reply:
> UPDATE: Changed to just default to fastfetch if it's installed.
> 
> I think that's a reasonable thing to do? If the user has installed fastfetch, they most likely want to use it. And if they run into a distro / ascii name mismatch, they can report so that we can add a workaround or something...

Definitly agree that it's reasonable; first thing I do on a fresh install is to 1. install hyfetch and 2. install fastfetch; I think most hyfetch users also use it.